prody/dynamics/adaptive.py

```diff
@@ -451,80 +393,59 @@ def runBothWaysAdaptiveANM(structureA, structureB, n_steps, **kwargs):
-    n_start = self.numSteps
+    n_modes0 = n_modes = kwargs.pop('n_modes', 20)
+
+    coordsA, coordsB, title, atoms, weights, rmsd = checkInput(structureA, structureB, **kwargs)
+    coordsA = coordsA.copy()
+    coordsB = coordsB.copy()
 
-    LOGGER.info('\n\nStarting from structure A ({0})'.format(self.titleA))
+    LOGGER.timeit('_prody_calcAdaptiveANM')
+    n = 0
     resetFmin = True
-    while self.numSteps < n_start + n_steps:
-        LOGGER.info('\nStarting cycle {0} with initial structure {1}'.format(self.numSteps+1,
-                                                                                self.titleA))
-        rmsd, new_coordsA, modesetA, modesCrossingFmin, self.dList, self.n_modes = _runStep(self.coordsA,
-                                                                                                       self.coordsB,
-                                                                                                       n_modes=self.n_modes,
-                                                                                                       resetFmin=resetFmin,
-                                                                                                       dList=self.dList,
-                                                                                                       **kwargs)
-        self.numSteps += 1
+    defvecs = []
+    rmsds = [rmsd]
+    ensA = Ensemble('A')
+    ensA.setCoords(coordsA)
+    ensA.setWeights(weights)
+    ensA.addCoordset(coordsA.copy())
+
+    ensB = Ensemble('B')
+    ensB.setCoords(coordsB.copy())
+    ensB.setWeights(weights)
+    ensB.addCoordset(coordsB.copy())
+    
+    while n < n_steps:
+        LOGGER.info('\nStarting cycle {0} with {1}'.format(n + 1, getTitle(structureA, 'structure A')))
+        n_modes = calcStep(coordsA, coordsB, n_modes, ensA, defvecs, rmsds,
+                           resetFmin=resetFmin, **kwargs)
+        n += 1
         resetFmin = False
-        self.anmListA.append(modesetA)
-        self.coordsA = new_coordsA
-        self.ensembleA.addCoordset(new_coordsA)
-        self.whichModesA.append(modesetA[modesCrossingFmin])
-        self.rmsds.append(rmsd)
-
-        LOGGER.debug('Total time so far is %.2f minutes' % (
-            (time.time() - LOGGER._times['_prody_runManySteps'])/60))
-        converged = _checkConvergence(
-            self.rmsds, self.coordsA, self.coordsB)
-        if converged:
-            # That way the original object is back to normal
-            #self.coordsA = self.coordsA_0
-            # That way the original object is back to normal
-            #self.coordsB = self.coordsB_0
-            LOGGER.warn('The part starting from structure A converged.')
-            break
 
-    LOGGER.info('\n\nStarting from structure B ({0})'.format(self.titleB))
+        if n_modes == 0:
+            LOGGER.report('Alternating Adaptive ANM converged in %.2fs.', '_prody_calcAdaptiveANM')
+            break
+    
+    n = 0
```

Comment:
I was expecting this to keep counting up rather than resetting so the total number of steps from both is used as a convergence criterion. This may work too. I guess it needs documenting carefully.

Reply:
The number of steps is not used as a convergence criterion currently. This needs to be reset otherwise `n<n_steps` can be `False` and structure B never got run. `n` is an internal variable for controlling the loop logic anyway.

Reply:
Perhaps you could change n_steps to reflect the steps that already took place.

Reply:
I think resetting `n` is simpler.

Reply:
No, I meant if you change `n_steps` then it reflects the reset to n. The new `n_steps` for B would be `n_steps_in - n_steps` for A if you reset `n`. If you don't reset `n` then `n_steps` stays as `n_steps_in`.